stac_fastapi/sqlalchemy/stac_fastapi/sqlalchemy/links.py

```diff
@@ -0,0 +1,57 @@
+"""Functionality to assist link construction."""
+
+import re
+from http.client import HTTP_PORT, HTTPS_PORT
+from urllib.parse import urljoin
+
+from starlette.requests import Request
+
+
+def get_base_url_from_request(request: Request) -> str:
+    """
+    Account for forwarding headers when deriving base URL.
+
+    Prioritise standard Forwarded header, look for non-standard X-Forwarded-* if missing.
+    Default to what can be derived from the URL if no headers provided.
+    """
+    if not isinstance(request, Request):
```

Comment:
This check doesn't exist in `pgstac`'s `get_base_url_from_request`?

Reply:
Good spot 😄  So far it's not necessary - `sqlalchemy` does a bunch of test setup with `MockStarletteRequest`, so all of its tests fail at this point without the `isinstance` check. `pgstac`'s setup is done differently, so the check isn't necessary.

Reply:
Ah... this https://github.com/sparkgeo/stac-fastapi/blob/9ce6895a50e3de27cf455646344d3e98dcfdbac6/stac_fastapi/sqlalchemy/tests/conftest.py#L166